iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -134,7 +137,10 @@ internal async Task<AmqpIotSession> EnsureSessionIsOpenAsync(CancellationToken c
             }
             catch (OperationCanceledException)
             {
-                throw new TimeoutException("Failed to enter the semaphore required for opening an AMQP session.");
+                throw new IotHubClientException(
```

Comment:
Why remap the exception? Seems we should let OperationCanceledException bubble up to the user.

Reply:
Yes, I blindly converted the original `TimeoutException` into `IotHubClientException` with new status Timeout here. I will throw `OperationCanceledException` as-is instead.